README.md

```diff
@@ -1,285 +1,366 @@
 
-	**Arguments:**
-	- _[OtcInterface](https://github.com/makerdao/maker-otc-support-methods/blob/master/src/MakerOtcSupportMethods.sol#L5) `otc`_ - An address of the contract that represents the OTC market contract. Only a few methods are exposed.
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `buyToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint buyAmt`_ - The amount that will be bought
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `wethToken`_ -The address of the `WETH token`
 
-	**Modifiers**
-	- _`*payable`_
+### Constructor
+**Summary:** This is used to inject the `WETH` token address which is used in some of the methods described below.
 
-	**Returns:**
-	- _`uint wethAmth`_ -the amount of `WETH token` that will be paid
+**Arguments**:
 
-	**Errors**
-	- _`otc.buyAllAmount`_ - might cause the TX to fail if the the caller is expected to pay more `ETH` than he sent.
-	- _`buyToken.transfer`_ - might cause the TX to fail
-	- _`withdrawAndSend`_ - might cause the TX to fail
+- `address wethToken` - The address of the [WETH](https://github.com/dapphub/ds-weth/blob/master/src/weth9.sol)  token. Note that it must be provided as a contract creation because the reference is needed in the `fallback` function.
 
-	---
 
-- **`buyAllAmountBuyEth`** - Used to exchange `ERC-20 token` for `ETH`. In the beginning the amount to be paid is calculated.   `payAmt` is transferred to `this` contract. After that the `otc` contract is ensured to have enough allowance in the `payToken` to do transfers on the behalf of `this` contract. Follows buy operation where `wethAmth` is bought. Last step includes transfer of the `wethAmt` to the caller.
+## Key functionalities (as defined in the Smart Contract)
 
-	**Arguments:**
-	- _[OtcInterface](https://githucalculated b.com/makerdao/maker-otc-support-methods/blob/master/src/MakerOtcSupportMethods.sol#L5) `otc`_ - An address of the contract that represents the OTC market contract. Only a few methods are exposed.
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `wethToken`_ -The address of the `WETH token`
-	 - _`uint wethAmt`_ - The amount that will be bought
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `payToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint maxPayAmt`_ - please check _*threshold_
+### createAndSellAllAmount
 
-	**Returns:**
-	-_`uint payAmt`_ - the amount of `payToken` that will be paid
+**Summary:** `createAndSellAllAmount`  is for `ERC-20 token` to `ERC-20 token` exchange. By calling this method, the sender creates a proxy and sells the exactly specified `ERC-20 token` amount.
```

Comment:
I think once we clarify that all of the methods in here just add proxy creation only , we wouldn't need summary for each of the methods ( it's just duplication ).